tests/integration/conftest.py

```diff
@@ -193,7 +193,10 @@ def add_constraints(self, constraints: Dict[str, str]):
             constraints:  Mapping of constraints to add to applications.
         """
         for app in self.applications.values():
-            val: str = app["constraints"]
+            if app.get("num_units", 0) < 1:
+                log.info("Skipping constraints for subordinate charm: %s", app["charm"])
+                continue
```

Comment:
prevent subordinate applications in our bundles `tests/data/*.yaml` from getting constraints added to them